tests/utils/test_timestamp.py

```diff
@@ -0,0 +1,161 @@
+import unittest
+from datetime import datetime
+
+from hypothesis import given
+from hypothesis import strategies as st
+from pytz import timezone
+
+from precog.utils.timestamp import (
+    datetime_to_iso8601,
+    datetime_to_posix,
+    get_midnight,
+    get_now,
+    get_posix,
+    get_timezone,
+    iso8601_to_datetime,
+    posix_to_datetime,
+)
+
+
+class TestTimestamp(unittest.TestCase):
+
+    # runs once prior to all tests in this file
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+        self.DATETIME_CONSTANT = datetime(2024, 12, 11, 18, 46, 43, 112378, tzinfo=get_timezone())
+        self.POSIX_CONSTANT = 1733942803.112378
+        self.ISO8601_CONSTANT = "2024-12-11T18:46:43.112378Z"
+
+    # runs once prior to every single test
+    def setUp(self):
+        pass
+
+    def test_get_timezone(self):
+        # Ensure we are abiding by UTC timezone
+        self.assertEqual(get_timezone(), timezone("UTC"))
+
+    def test_get_now(self):
+        now = get_now()
+
+        # Check that this is a datetime
+        self.assertIsInstance(now, datetime)
+
+        # Check that this is UTC timezone aware
+        self.assertEqual(now.tzinfo, get_timezone())
+
+        # Check that the timestamp is sensitive
+        now2 = get_now()
+        self.assertNotEqual(now, now2)
+
+        # Make our own timestamp
+        real_now = datetime.now(get_timezone())
+
+        # Check that all timezones are within 20 seconds of each other
+        # They should all be very close together considering everything
+        threshold = 20
+        diff1 = (now - now2).total_seconds()
+        diff2 = (now - real_now).total_seconds()
+        diff3 = (now2 - real_now).total_seconds()
+
+        self.assertLess(abs(diff1), threshold)
+        self.assertLess(abs(diff2), threshold)
+        self.assertLess(abs(diff3), threshold)
+
+    def test_get_posix(self):
+        posix = get_posix()
+
+        # Check that this is a float
+        self.assertIsInstance(posix, float)
+
+        # Check that the timestamp is sensitive
+        posix2 = get_posix()
+        self.assertNotEqual(posix, posix2)
+
+        # Make our own timestamp
+        real_posix = datetime.now(get_timezone()).timestamp()
+
+        # Check that all timezones are within 20 seconds of each other
+        # They should all be very close together considering everything
+        threshold = 20
+        diff1 = posix - posix2
+        diff2 = posix - real_posix
+        diff3 = posix2 - real_posix
+
+        self.assertLess(abs(diff1), threshold)
+        self.assertLess(abs(diff2), threshold)
+        self.assertLess(abs(diff3), threshold)
+
+    def test_get_midnight(self):
+        now = datetime.now(get_timezone())
+        midnight = now.replace(hour=0, minute=0, second=0, microsecond=0)
```

Comment:
any reason to do now.replace() rather than just the datetime() constructor?

Reply:
i don't know if there's an easy way to do this without getting the current timestamp. with the datetime constructor, we would need to get the current year / month / day. whereas the current code works in the opposite direction.

Reply:
Oh I see you still want to refer to the current date, just with 00:00:00 time.  Yeah makes sense, the only alternative would be to test a static date like datetime(year=2024, month=12, day=1); but I think your approach is good.  Maybe I just didnt understand what midnight was for